src/Api/Serializer/RestSerializer.php

```diff
@@ -126,6 +126,13 @@ private function applyHeader($name, Shape $member, $value, array &$opts)
         if ($member->getType() == 'timestamp') {
             $value = TimestampShape::format($value, 'rfc822');
         }
+        if ($member['jsonvalue']) {
+            $value = json_encode($value);
+            if ($value === null && json_last_error() !== JSON_ERROR_NONE) {
+                throw new \InvalidArgumentException('Unable to encode the provided value as JSON');
```

Comment:
Perhaps we could call out the fact that we are using `json_encode`, which could help debugging if this cause troubles for some customer. Such as:
`Unable to encode the provided value with 'json_encode'.`